crates/bevy_gizmos/src/lib.rs

```diff
@@ -131,31 +131,42 @@ fn update_gizmo_meshes(
 ) {
     let list_mesh = meshes.get_mut(&handles.list).unwrap();
 
-    let positions = mem::take(&mut storage.list_positions);
-    list_mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, positions);
+    storage.in_use = false;
 
-    let colors = mem::take(&mut storage.list_colors);
-    list_mesh.insert_attribute(Mesh::ATTRIBUTE_COLOR, colors);
+    if !storage.list_positions.is_empty() {
+        storage.in_use = true;
 
-    let strip_mesh = meshes.get_mut(&handles.strip).unwrap();
+        let positions = mem::take(&mut storage.list_positions);
+        list_mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, positions);
 
-    let positions = mem::take(&mut storage.strip_positions);
-    strip_mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, positions);
+        let colors = mem::take(&mut storage.list_colors);
+        list_mesh.insert_attribute(Mesh::ATTRIBUTE_COLOR, colors);
+    }
+
+    if !storage.strip_positions.is_empty() {
+        storage.in_use = true;
```

Comment:
Instead of setting this flag, shouldn't we just check if either positions are empty? It'd avoid the extra field.